src/android/SecureStorage.java

```diff
@@ -41,39 +42,65 @@ protected void pluginInitialize() {
     private boolean isDeviceSecure() {
         KeyguardManager keyguardManager = (KeyguardManager) (getContext().getSystemService(Context.KEYGUARD_SERVICE));
         try {
+
+            // This tries to call isDeviceSecure, which was only added in API 23
+            // The method checks if there is a lock screen that requires authentication defined (not None or Swipe)
+            // This is preferred to the older method isKeyguardSecure, that also returns true if the SIM card is unlocked
             Method isSecure = null;
             isSecure = keyguardManager.getClass().getMethod("isDeviceSecure");
             return ((Boolean) isSecure.invoke(keyguardManager)).booleanValue();
+
         } catch (Exception e) {
+
+            // Best effort if the preferred method is unavailable
```

Comment:
Shouldn't there be a log in this `catch` block?

Reply:
We can but from what I could gather this will only throw if the method does not exist, i.e. a reflection exception. In this case the catch is running the expected case.

This code is using reflection so it can be compiled with targets lower than 23. We only build with targets of 26+, so we can refactor this code to not use reflection at all and instead use a version check. But this is outside the scope of this issue, and since the whole plugin is in the dire need of a refactor I'd prefer to keep this change to the eventual refactor.

Reply:
Fair enough. Good call.